### ISSUE TITLE
failed to build acrn at ubuntu16.04 + gcc-5

### ISSUE THREAD

**Issue by zhijianli88** [projectacrn/acrn-hypervisor#480]
try to bisect, and figure out the FBC
```
lizj@G08FNSTD140138:~/workspace/Audio/acrn-hypervisor$ git bisect log 
lizj@G08FNSTD140138:~/workspace/Audio/acrn-hypervisor$ git bisect log 
git bisect start
# good: [32fccb2f43671366266c61049c73757a6dac8b67] HV: 'vlapic_set_local_intr()' code cleanup
git bisect good 32fccb2f43671366266c61049c73757a6dac8b67
# good: [32fccb2f43671366266c61049c73757a6dac8b67] HV: 'vlapic_set_local_intr()' code cleanup
git bisect good 32fccb2f43671366266c61049c73757a6dac8b67
# bad: [804026168c208077c00b058102b14cfa856e47de] HV: remove unused API lookp_entry_by_id()
git bisect bad 804026168c208077c00b058102b14cfa856e47de
# bad: [d705970eb268bf03ed47e20951d6b5d7cdc234ac] HV: vmx: convert hexadecimals used in bitops to unsigned
git bisect bad d705970eb268bf03ed47e20951d6b5d7cdc234ac
# good: [5aca8b1cdd41f7f1f23e990914c7df5dd4dc0f73] When guest frees vector, counts needs to be reset. "int" command from hv shell ends up incrementing per cpu irq_count.
git bisect good 5aca8b1cdd41f7f1f23e990914c7df5dd4dc0f73
# bad: [f4bd0798e0eb89c133efe53471ba284d73b73ba4] HV: mmu: convert hexadecimals used in bitops to unsigned
git bisect bad f4bd0798e0eb89c133efe53471ba284d73b73ba4
# bad: [7b548e87db1b9c60eb927b93cfb30449bb4d6b88] HV: cpu: convert hexadecimals used in bitops to unsigned
git bisect bad 7b548e87db1b9c60eb927b93cfb30449bb4d6b88
# first bad commit: [7b548e87db1b9c60eb927b93cfb30449bb4d6b88] HV: cpu: convert hexadecimals used in bitops to unsigned
```
i use ubuntu 16.04 and gcc-5
```
lizj@G08FNSTD140138:~/workspace/Audio/acrn-hypervisor$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 5.4.0-6ubuntu1~16.04.9' --with-bugurl=file:///usr/share/doc/gcc-5/README.Bugs --enable-languages=c,ada,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-5 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-5-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-5-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-5-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.9) 
```

**Comment by zhijianli88** [comment]
attached the failures
···
[ ! -e /home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/arch/x86/trampoline.o ] && mkdir -p /home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/arch/x86/; \
cc  -Iinclude  -Iinclude/lib  -Iinclude/lib/crypto  -Iinclude/common  -Iinclude/arch/x86  -Iinclude/arch/x86/guest  -Iinclude/debug  -Iinclude/public  -Iinclude/common  -Ibsp/include  -Ibsp/uefi/include/bsp  -Iboot/include  -I/home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/include -I. -m64 -nostdinc -nostdlib -gdwarf-2 -DASSEMBLER=1 -c arch/x86/trampoline.S -o /home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/arch/x86/trampoline.o
arch/x86/trampoline.S: Assembler messages:
arch/x86/trampoline.S:81: Error: missing ')'
arch/x86/trampoline.S:81: Error: missing ')'
arch/x86/trampoline.S:81: Error: junk `U<<3)|(1U<<5)|(1U<<6)|(1U<<9)|(1U<<10))' after expression
arch/x86/trampoline.S:92: Error: junk `U' after expression
arch/x86/trampoline.S:94: Error: missing ')'
arch/x86/trampoline.S:94: Error: junk `U<<8)' after expression
arch/x86/trampoline.S:101: Error: missing ')'
arch/x86/trampoline.S:101: Error: missing ')'
arch/x86/trampoline.S:101: Error: junk `U<<31)|(1U<<0)|(1U<<1)|(1U<<5))' after expression
arch/x86/trampoline.S:146: Error: junk `UL' after expression
arch/x86/trampoline.S:150: Warning: indirect jmp without `*'
arch/x86/trampoline.S:183: Error: missing ')'
arch/x86/trampoline.S:183: Error: junk at end of line, first unrecognized character is `U'
arch/x86/trampoline.S:192: Error: missing ')'
arch/x86/trampoline.S:192: Error: junk at end of line, first unrecognized character is `U'
arch/x86/trampoline.S:192: Error: missing ')'
arch/x86/trampoline.S:192: Error: junk at end of line, first unrecognized character is `U'
...
arch/x86/trampoline.S:199: Error: missing ')'
arch/x86/trampoline.S:199: Error: junk at end of line, first unrecognized character is `U'
arch/x86/trampoline.S:199: Error: missing ')'
arch/x86/trampoline.S:199: Error: junk at end of line, first unrecognized character is `U'
Makefile:256: recipe for target '/home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/arch/x86/trampoline.o' failed
make[1]: *** [/home/lizj/workspace/Audio/acrn-hypervisor/build/hypervisor/arch/x86/trampoline.o] Error 1
make[1]: Leaving directory '/home/lizj/workspace/Audio/acrn-hypervisor/hypervisor'
Makefile:20: recipe for target 'hypervisor' failed
make: *** [hypervisor] Error 2
···

**Comment by junjiemao1** [comment]
Hi @zhijianli88 , thanks for reporting this. The root cause is the assembler not supporting C-style unsigned constants like 0xffU (refer to https://sourceware.org/binutils/docs/as/Integers.html#Integers for the syntax of integers supported by binutils as).

Though not officially supported in the manual, I find the assembler in binutils >= 2.28 understands such constants while binutils 2.26 (shipped with Ubuntu 16.04) does not. You can either upgrade binutils or pick #485 which fixes this issue.
